packages/tooling/fast-tooling/src/web-components/html-render-layer-navigation/html-render-layer-navigation.ts

```diff
@@ -35,9 +48,18 @@ export class HTMLRenderLayerNavigation extends HTMLRenderLayer {
     private timeoutRef: number = null;
     private currElementRef: HTMLElement = null;
 
+    //TODO: Replace with vanila ResizeObserver after Typscript is upgraded to at least v4.2.3.
+    private resizeDetector: ResizeObserverClassDefinition | null = null;
+
     connectedCallback() {
         super.connectedCallback();
 
+        this.resizeDetector = new ((window as unknown) as WindowWithResizeObserver).ResizeObserver(
+            this.handleWindowChange
+        );
+
+        this.resizeDetector.observe(document.body);
```

Comment:
This seems like it will be called ALOT during page loading. We might want to think about debouncing handleWindowChange().

Reply:
There is a debouncer. Line 85 starts the setTimeout. Also the update logic only happens when an element is actively highlighted or selected which does not occur until until after the markup is rendered.